### PR TITLE
Expand dba rules

### PR DESCRIPTION
**Pull request by rroutsong**
Code to address #43 

Split diff bind into a stack of multiple rules including:

A parent rule for creating the inputs for diff bind

A child rule for running diff bind without blocking

A child rule for running diff bind with blocking

I also applied this format to the uropa differential binding analysis rules.

Validation testing is under-way at `/data/OpenOmics/dev/datasets/outputs`

### CODE REVIEW

**Review by skchronicles — APPROVED** [review]
Thank you for splitting up the uropa rules and adding the timestamp to the snakemake log file

File: workflow/rules/dba.smk [hunk 1]
```diff
@@ -36,139 +37,253 @@ otherDirs                       = [qc_dir, homer_dir, uropa_dir]
...
         this_peakextension              = lambda w: PeakExtensions[w.PeakTool],
         peakcaller                      = lambda w: FileTypesDiffBind[w.PeakTool],
-        # scripts in the bin directory used in the shell directive
         rscript                         = join(bin_path, "DiffBind_v2_ChIPseq.Rmd"),
-        pythonscript                    = join(bin_path, "prep_diffbind.py"),
-        blocking_rscript                = join(bin_path, "DiffBind_v2_ChIPseq_block.Rmd"),
-        # output base directories or full file locations
         outdir                          = join(diffbind_dir, "{group1}_vs_{group2}-{PeakTool}"),
-        csvfile                         = join(
+    containerized:
```
[the remark this answers is not in the document]
Comment (rroutsong) [reply workflow/rules/dba.smk]: this is not the latest iteration of changes, in the most recent commit they are all updated

File: config/containers.json [hunk 2]
```diff
@@ -3,5 +3,9 @@
         "cfchip": "docker://skchronicles/cfchip_toolkit:v0.5.0",
         "python": "docker://asyakhleborodova/chrom_seek_python:v0.1.0",
         "ppqt": "docker://asyakhleborodova/ppqt:v0.2.0"
+    }, 
+    "containers": {
```
[the remark this answers is not in the document]
Comment (rroutsong) [reply config/containers.json:8]: it will always pull images if you do not specify a `--sif-cache` we can push this issue to another PR

**Review by tovahmarkowitz — APPROVED** [review]
Looks good!

**Pull request merged**